### PR TITLE
elfobj.d: use Rarray for SegData

### PR DESCRIPTION
**Pull request by WalterBright**
Rarray does it better.

### CODE REVIEW

**Review by welkam — COMMENTED** [review]
Dont know how much weight my review carries but still

File: src/dmd/backend/elfobj.d [hunk 1]
```diff
@@ -299,6 +300,9 @@ int         MAP_SEG2TYP(int seg)    { return MAP_SEG2SEC(seg).sh_flags & SHF_EXE
 public seg_data **SegData;
 public int seg_length;          // used length of SegData[]
```
Comment (MoonlightSentinel) [inline src/dmd/backend/elfobj.d:301]: These variables seem to be redundant to SegDataR now?

Comment (welkam) [reply src/dmd/backend/elfobj.d:301]: They are still used in the file and probably will be removed by following pull requests. Might be useful to move all SegData refactoring to separate branch

Comment (WalterBright) [reply src/dmd/backend/elfobj.d:301]: Right. Messing with memory allocation is tricky and I want to minimize diffs while doing that.

Comment (MoonlightSentinel) [reply src/dmd/backend/elfobj.d:301]: Fair enough

**Pull request merged**